elide-core/src/main/java/com/yahoo/elide/security/FilterExpressionCheck.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2016, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.security;
+
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.filter.Predicate;
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.parsers.expression.FilterExpressionCheckOkVisitor;
+import com.yahoo.elide.security.checks.InlineCheck;
+
+import java.util.Optional;
+
+/**
+ * Check for FilterExpression. This provide an super class for user defined FilterExpression check. The subclass should
+ * override getFilterExpression function and return a FilterExpression which will be passed down to datastore.
+ * @param <T> Type of class
+ */
+public abstract class FilterExpressionCheck<T> extends InlineCheck<T> {
+
+    public abstract FilterExpression getFilterExpression(RequestScope requestScope);
+
+    @Override
+    public final boolean ok(T object, RequestScope requestScope, Optional<ChangeSpec> changeSpec) {
```

Comment:
Add a comment that the filter expression is evaluated in memory if it cannot be pushed to the data store by elide for any reason.